src/sep8/getApprovalServerUrl.test.ts

```diff
@@ -0,0 +1,182 @@
+import axios from "axios";
+import sinon from "sinon";
+import { Config } from "stellar-sdk";
+import { getApprovalServerUrl } from "./getApprovalServerUrl";
+
+describe("getApprovalServerUrl", () => {
+  let axiosMock: sinon.SinonMock;
+
+  beforeEach(() => {
+    axiosMock = sinon.mock(axios);
+    Config.setDefault();
+  });
+
+  afterEach(() => {
+    axiosMock.verify();
+    axiosMock.restore();
+  });
+
+  test("Issuer's Home Domain missing", async () => {
+    try {
+      // @ts-ignore
+      const res = await getApprovalServerUrl({
+        asset_code: "USD",
+        asset_issuer:
+          "GDBMMVJKWGT2N6HZ2BGMFHKODASVFYIHL2VS3RUTB3B3QES2R6YFXGQW",
+      });
+      expect("This test failed").toBe(null);
+    } catch (e) {
+      expect(e.toString()).toMatch(`Error: Issuer's home domain is missing`);
+    }
+  });
+
+  test("stellar.toml CURRENCIES missing", async () => {
+    const homeDomain = "example.com";
+    axiosMock
+      .expects("get")
+      .withArgs(sinon.match(`https://${homeDomain}/.well-known/stellar.toml`))
+      .returns(
+        Promise.resolve({
+          data: "",
+        }),
+      );
+
+    try {
+      // @ts-ignore
+      const res = await getApprovalServerUrl({
+        asset_code: "USD",
+        asset_issuer:
+          "GDBMMVJKWGT2N6HZ2BGMFHKODASVFYIHL2VS3RUTB3B3QES2R6YFXGQW",
+        home_domain: homeDomain,
+      });
+      expect("This test failed").toBe(null);
+    } catch (e) {
+      expect(e.toString()).toMatch(
+        `Error: stellar.toml at ${homeDomain} does not contain CURRENCIES` +
+          ` field`,
+      );
+    }
+  });
+
+  test("stellar.toml CURRENCIES missing", async () => {
+    const homeDomain = "example.com";
+    axiosMock
+      .expects("get")
+      .withArgs(sinon.match(`https://${homeDomain}/.well-known/stellar.toml`))
+      .returns(
+        Promise.resolve({
+          data: "",
+        }),
+      );
+
+    try {
+      // @ts-ignore
+      const res = await getApprovalServerUrl({
+        asset_code: "USD",
+        asset_issuer:
+          "GDBMMVJKWGT2N6HZ2BGMFHKODASVFYIHL2VS3RUTB3B3QES2R6YFXGQW",
+        home_domain: homeDomain,
+      });
+      expect("This test failed").toBe(null);
+    } catch (e) {
+      expect(e.toString()).toMatch(
+        `Error: stellar.toml at ${homeDomain} does not contain CURRENCIES` +
+          ` field`,
+      );
+    }
+  });
```

Comment:
This test looks like a duplicate (one above).

Reply:
Ah you're right 🤦 https://github.com/stellar/js-stellar-wallets/pull/210/commits/c9fb11a3962072e4cf4d9cb5715d271391a318df